### PR TITLE
containerdb

### PR DESCRIPTION
**Pull request by alfonsoros88**
The intention is to simplify a bit the source of state.rs by moving to a new module all the logic for managing containers and repositories.

### CODE REVIEW

File: northstar/src/runtime/containerdb.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        }
+
+        self.find_npk(container).map(ManifestRef)
+    }
+
+    /// Find the repository that has the given container
+    pub fn find_repository(&self, container: &Container) -> Option<&dyn Repository> {
+        self.index_by_container
+            .get(container)
+            .and_then(|&idx| self.repositories.get(idx).map(|(_, b)| b.as_ref()))
```
Comment (flxo) [inline northstar/src/runtime/containerdb.rs:84]: Index operations scare me...

**Review by flxo — COMMENTED** [review]
The relation between the lines added and removed or simplified by this patch is a little bit odd. 
The improvements in the `state` module where NPKs or repositories are searched are kind of small and the complexity with the patch is almost the same as before.

File: northstar/src/runtime/containerdb.rs (hunk 1 above)
Comment (alfonsoros88) [reply northstar/src/runtime/containerdb.rs:84]: i'd agree, on the other hand this is not unsafe, should not panic if there is a bug,

**Comment by alfonsoros88** [comment]
> The relation between the lines added and removed or simplified by this patch is a little bit odd. The improvements in the `state` module where NPKs or repositories are searched are kind of small and the complexity with the patch is almost the same as before.

Well yes, obviously there are more insertions due to the new module with its unit tests and so on. There is not much simplification in terms of changing things mainly moving and separating stuff. The intention here is to move the logic that wasn't inherently a responsibility of state. I see this as the first step and hopefully in subsequent patches we can better organize this part of the code. I think that the new module makes sense and its kind of clear what are its responsibilities.

**Comment by alfonsoros88** [comment]
closing this one in favor o #438

**Pull request closed without merging**